File: 02-intro-ITK-codigo/C++/02_multi_io.cxx
#include <iostream>

#include <itkImage.h>
#include <itkImageSeriesReader.h>
#include <itkImageFileWriter.h>

// -------------------------------------------------------------------------
const unsigned int Dimension = 3;
typedef unsigned char TPixel;  // [0,255]
/*
  typedef unsigned short TPixel; // [0,65535]
  typedef unsigned int TPixel;   // [0,4294967295]
  typedef unsigned long TPixel;  // [0,18446744073709551615]
  typedef char TPixel;           // [-128,127]
  typedef short TPixel;          // [-32768, 32767]
  typedef int TPixel;            // [-2147483648, 2147483647]
  typedef long TPixel;           // [-9223372036854775808, 9223372036854775807]
  typedef float TPixel;          // [1.17549e-38, 3.40282e+38]
  typedef double TPixel;         // [2.22507e-308, 1.79769e+308]
*/
typedef itk::Image< TPixel, Dimension >  TImage;
typedef itk::ImageSeriesReader< TImage > TReader;
typedef itk::ImageFileWriter< TImage >   TWriter;

// -------------------------------------------------------------------------
int main( int argc, char* argv[] )
{
  // Avoid annoying messages
  itk::Object::GlobalWarningDisplayOff( );

  // Read series
  TReader::Pointer reader = TReader::New( );
  for( int i = 1; i < argc - 1; i++ )
    reader->AddFileName( argv[ i ] );

  // Write single image
  TWriter::Pointer writer = TWriter::New( );
  writer->SetInput( reader->GetOutput( ) );
  writer->SetFileName( argv[ argc - 1 ] );
  try
  {
    writer->Update( );
  }
  catch( itk::ExceptionObject& err )
  {
    std::cerr << "An error has occurred: " << err << std::endl;
    return( 1 );

  } // yrt

  return( 0 );
}

// eof - 02_multi_io.cxx

File: 02-intro-ITK-codigo/Python/02_multi_io.py
import itk
import sys

# -------------------------------------------------------------------------
# Base image
Dimension = 3 
TPixel = itk.UC  # [0,255]

# TPixel = itk.US  # [0,65535]
# TPixel = itk.UI  # [0,4294967295]
# TPixel = itk.UL  # [0,18446744073709551615]
# TPixel = itk.C   # [-128,127]
# TPixel = itk.S   # [-32768, 32767]
# TPixel = itk.I   # [-2147483648, 2147483647]
# TPixel = itk.L   # [-9223372036854775808, 9223372036854775807]
# TPixel = itk.F   # [1.17549e-38, 3.40282e+38]
# TPixel = itk.D   # [2.22507e-308, 1.79769e+308]

TImage = itk.Image[TPixel, Dimension]

# -------------------------------------------------------------------------
# Read series
reader = itk.ImageSeriesReader[TImage].New( )
for i in range(1, len(sys.argv) - 1) :
    reader.AddFileName( sys.argv[i] )

# Write single image
writer = itk.ImageFileWriter[TImage].New( )
writer.SetInput( reader.GetOutput( ) )
writer.SetFileName( sys.argv[ len(sys.argv) - 1 ] )
writer.Update()

# eof - 02_multi_io.py
